tests/test_code_actions.py

```diff
@@ -232,25 +232,35 @@ def _setup_document_with_missing_semicolon(self) -> Generator:
 
 class CodeActionMatchingTestCase(unittest.TestCase):
     def test_does_not_match(self) -> None:
-        actual = get_matching_kinds({'a.x': True}, ['a.b'])
+        actual = get_matching_on_save_kinds({'a.x': True}, ['a.b'])
         expected = []  # type: List[str]
         self.assertEquals(actual, expected)
 
     def test_matches_exact_action(self) -> None:
-        actual = get_matching_kinds({'a.b': True}, ['a.b'])
+        actual = get_matching_on_save_kinds({'a.b': True}, ['a.b'])
         expected = ['a.b']
         self.assertEquals(actual, expected)
 
     def test_matches_more_specific_action(self) -> None:
-        actual = get_matching_kinds({'a.b': True}, ['a.b.c'])
+        actual = get_matching_on_save_kinds({'a.b': True}, ['a.b.c'])
         expected = ['a.b.c']
         self.assertEquals(actual, expected)
 
     def test_does_not_match_disabled_action(self) -> None:
-        actual = get_matching_kinds({'a.b': True, 'a.b.c': False}, ['a.b.c'])
+        actual = get_matching_on_save_kinds({'a.b': True, 'a.b.c': False}, ['a.b.c'])
         expected = []  # type: List[str]
         self.assertEquals(actual, expected)
 
+    def test_kind_matching(self) -> None:
+        self.assertTrue(kinds_include_kind(['a'], 'a.b'))
+        self.assertFalse(kinds_include_kind(['a'], 'b.a'))
+        self.assertFalse(kinds_include_kind(['a.b'], 'b'))
+        self.assertTrue(kinds_include_kind(['a.b'], 'a.b'))
+        self.assertFalse(kinds_include_kind(['a.b'], 'a'))
+        self.assertFalse(kinds_include_kind(['aa'], 'a'))
+        self.assertFalse(kinds_include_kind(['aa.b'], 'a'))
+        self.assertFalse(kinds_include_kind(['aa.b'], 'b'))
```

Comment:
This reminds me a lot of the `sublime.score_selector` function. Is it equivalent?

Reply:
It does indeed seem to work in the same way if we transform first argument to space-separate string. I guess using it directly might be confusing but having a wrapper function with more relevant name should be fine...

Reply:
Actually maybe not since it's "backwards".

`scope_selector` matches single selectors against many scopes while we need to match many selectors against a single scope.

We could call it separately for each selector (item in `kinds`) but that seem a bit wasteful.